src/components/donation-flow/DonationFlowForm.tsx

```diff
@@ -152,32 +171,54 @@ export function DonationFlowForm() {
               }}
               autoComplete="off">
               <Box mb={2}>
-                <StepSplitter content="1" active={Boolean(values.amount)} />
+                <StepSplitter content="1" active={Boolean(values.amountChosen)} />
                 <Amount />
                 <StepSplitter
                   content="2"
-                  active={Boolean(values.amount) && Boolean(values.payment)}
+                  active={Boolean(values.amountChosen) && Boolean(values.payment)}
                 />
-                <PaymentMethod />
+                <PaymentMethod sectionRef={paymentMethodSectionRef} />
                 <StepSplitter
                   content="3"
                   active={
-                    Boolean(values.amount) &&
+                    Boolean(values.amountChosen) &&
                     Boolean(values.payment) &&
                     Boolean(values.authentication)
                   }
                 />
-                <Authentication />
+                <Authentication sectionRef={authenticationSectionRef} />
               </Box>
+              <StepSplitter />
+              <CheckboxField
+                label={
+                  <Box display="flex" alignItems="center">
+                    <Typography>Искам да съм анонимен</Typography>
+                    <Tooltip title="Ако дарете анонимно, данните ще останат недостъпни за бенефициента.">
+                      <IconButton color="primary">
+                        <Info />
+                      </IconButton>
+                    </Tooltip>
+                  </Box>
+                }
+                name="isAnonymous"
+              />
+              {JSON.stringify(paymentError)}
```

Comment:
We might need to take this off in the final version. Mentioning it so we don't forget it

```suggestion
```